Wox.Core/Configuration/Portable.cs

```diff
@@ -0,0 +1,200 @@
+using Microsoft.Win32;
+using Squirrel;
+using System;
+using System.IO;
+using System.Reflection;
+using System.Windows;
+using Wox.Infrastructure;
+using Wox.Infrastructure.Logger;
+using Wox.Infrastructure.UserSettings;
+using Wox.Plugin.SharedCommands;
+
+namespace Wox.Core.Configuration
+{
+    public class Portable : IPortable
+    {
+        private UpdateManager portabilityUpdater;
+
+        public void DisablePortableMode()
+        {
+            portabilityUpdater = new UpdateManager(string.Empty, Constant.Wox, Constant.RootDirectory);
+
+            try
+            {
+                MoveUserDataFolder(DataLocation.PortableDataPath, DataLocation.RoamingDataPath);
+#if DEBUG
+                // Create shortcuts and uninstaller are not required in debug mode, 
+                // otherwise will repoint the path of the actual installed production version to the debug version
+#else
+                CreateShortcuts();
+                CreateUninstallerEntry();
+#endif
+                IndicateDeletion(DataLocation.PortableDataPath);
+
+                MessageBox.Show("Wox needs to restart to finish disabling portable mode, " +
+                    "after the restart your portable data profile will be deleted and roaming data profile kept");
+
+                portabilityUpdater.Dispose();
+
+                UpdateManager.RestartApp();
+            }
+            catch (Exception e)
+            {
+#if DEBUG
+                portabilityUpdater.Dispose();
+                throw;
+#else
+                portabilityUpdater.Dispose();
+                Log.Exception("Portable", "Error occured while disabling portable mode", e);
```

Comment:
Maybe something like this?
portabilityUpdater.Dispose();
#if !DEBUG
                Log.Exception("Portable", "Error occured while disabling portable mode", e);
#endif
throw;